src/main/java/hudson/remoting/Channel.java

```diff
@@ -829,8 +830,16 @@ public void terminate(IOException e) {
             } // JENKINS-14909: leave synch block
         } finally {
             if (e instanceof OrderlyShutdown) e = null;
-            for (Listener l : listeners.toArray(new Listener[0]))
-                l.onClosed(this, e);
+            for (Listener l : listeners.toArray(new Listener[0])) {
+                try {
+                    l.onClosed(this, e);
+                } catch (Throwable t) {
```

Comment:
The point being that when the listener is being called, the channel has actually been closed, so we don't want to log a failure to close where it is actually a fault of the listener